src/emcpy/plots/mapping.py

```diff
@@ -0,0 +1,130 @@
+        plt.grid(plotopts['grid'])
+
+    cax = fig.add_axes([ax.get_position().x1 + 0.02,
+                        ax.get_position().y0, 0.025,
+                        ax.get_position().height])
+
+    cb = plt.colorbar(cs, extend='both', cax=cax)
+    cb.set_label(plotopts['cbar label'])
+
+    ax.set_title(plotopts['title'], loc='left')
+    ax.set_title(plotopts['time title'], loc='right', fontweight='semibold')
+
+    return fig
+
+
+def _map_pcolormesh(latitude, longitude, data, domain, plotmap, plotopts):
+    """
+    Plot 2-dimensional data on a map as pcolormesh.
+    """
+    fig = plt.figure(figsize=plotopts['figsize'])
+
+    if plotmap:
+        ax = domains.get_domain(fig, domain)
+        cs = ax.pcolormesh(longitude, latitude, data, cmap=plotopts['cmap'],
+                           vmin=plotopts['vmin'], vmax=plotopts['vmax'],
+                           transform=ccrs.PlateCarree())
+        if plotopts['grid']:
+            ax.gridlines(crs=ccrs.PlateCarree())
+
+    else:
+        ax = fig.add_subplot()
+        cs = plt.pcolormesh(longitude, latitude, data, cmap=plotopts['cmap'],
+                            vmin=plotopts['vmin'], vmax=plotopts['vmax'])
+        plt.grid(plotopts['grid'])
+
+    cax = fig.add_axes([ax.get_position().x1 + 0.02,
+                        ax.get_position().y0, 0.025,
+                        ax.get_position().height])
+
+    cb = plt.colorbar(cs, extend='both', cax=cax)
+    cb.set_label(plotopts['cbar label'])
+
+    ax.set_title(plotopts['title'], loc='left')
+    ax.set_title(plotopts['time title'], loc='right', fontweight='semibold')
+
+    return fig
+
+
+def map2d(latitude, longitude, data, domain='global', plotmap=True,
+          figsize=(10, 8), cmap='viridis', markersize=10, grid=False,
+          vmin=None, vmax=None, title='EMCPy Map Plot', time_title=None,
+          cbar_label=None):
+    """
+    Plot a spatial map of desired data.
+
+    Args:
+        latitude : (array type) Latitude array for spatial data
+        longitude : (array type) Longitude array for spatial data
+        data : (array type) Spatial data to be plotted
+        domain : (str, default='global') Map domain for plot
+        plotmap : (bool, default=True) Plots data with a background map
+        figsize : (tuple, default=(10,8)) Figure size
+        cmap : (str, optional, default='viridis') Color map of data
+        markersize : (int, optional, default=10) Marker size for 1D data types
+        grid : (bool, optional, default=False) Plot grid on spatial plot
+        vmin : (int, optional, default=None) Minimum value displayed on plot
+        vmax : (int, optional, default=None) Maximum value displayed on plot
+        title : (str, optional, default='EMCPy Map Plot') Plot title
+        time_title : (str, optional, default=None) Secondary title for
+                     date/cycle to add to plot
+        cbar_label : (str, optional, deafult=None) Colorbar label
```

Comment:
last thing, cbar_label doesn't work (is it off the figure page?)

Reply:
Really? When I test it still works.
```
fig = emcpy.plots.map2d(lat, lon, data2, domain='conus', plotmap=True, cmap='magma',
                        grid=False, title="Test Plot Using Random 2D Data",
                        time_title="July 27, 2021 3:19pm",
                        cbar_label="random values")
```

![image](https://user-images.githubusercontent.com/69815622/127329855-5b37b00c-ce04-470f-b7a4-5ad9fb2ae0c7.png)